src/test/java/org/broadinstitute/hellbender/utils/variant/GATKVariantContextUtilsUnitTest.java

```diff
@@ -2206,4 +2346,106 @@ public void testMatchAlleles(final SimpleInterval variant1Interval, final List<S
         final int[] matches = GATKVariantContextUtils.matchAllelesOnly(variant1, variant2);
         Assert.assertTrue(Arrays.equals(matches, gtMatch), "Failed");
     }
+
+    @DataProvider(name = "multiAllelic")
+    public Object[][] multiAllelicDataProvider() {
```

Comment:
If you decide that you really do want to keep both codepaths for splitting alleles, i would at least extract out some way to test both using the tests in `AlleleSubsettingUtils`